app/controllers/files/download_controller.rb

```diff
@@ -1,6 +1,15 @@
 class Files::DownloadController < Files::FilesController
   skip_before_filter :set_identity
-  before_filter :set_file, only: :download_url
+  before_filter :set_file, except: :download
+
+  def preview
+    if @file.identity.user == current_user
+      # send the file to the user
+      send_data @file.identity.get_file(@file.foreign_ref), disposition: 'inline',
```

Comment:
Created a [separate branch](https://github.com/tkomstadius/bruse/tree/feature/view-files-cached) for this. Sorry for wasting my time with something that isn't really needed.